### PR TITLE
Fix off-by-one bug in Blockstore::purge_exact()

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
`to_slot` is incremented by one in order to make the loop run for `[from_slot, to_slot]`. However, this value is then used with a `<=` later on; we need the original (non-incremented) value for the comparison to be correct.

#### Summary of Changes
Use the incremented value in the loop bounds only instead of updating `to_slot`.

This could also be done with `from_slot..=to_slot`; I'm guessing `saturating_add(1)` was used to try to make the inclusivity more obvious. I could go either way on using `..=` vs `.saturating_add(1)`

### CODE REVIEW

**Review by CriesofCarrots — APPROVED** [review]
Good catch, thanks!

**Comment by yhchiang-sol** [comment]
Reviewing.

File: ledger/src/blockstore/blockstore_purge.rs [hunk 1]
```diff
@@ -354,8 +354,7 @@ impl Blockstore {
     ) -> Result<()> {
         let mut index0 = self.transaction_status_index_cf.get(0)?.unwrap_or_default();
         let mut index1 = self.transaction_status_index_cf.get(1)?.unwrap_or_default();
-        let to_slot = to_slot.saturating_add(1);
-        for slot in from_slot..to_slot {
+        for slot in from_slot..to_slot.saturating_add(1) {
```
Comment (yhchiang-sol) [inline ledger/src/blockstore/blockstore_purge.rs:357]: How about we simply use `..=`?

    for slot in from_slot..=to_slot {

**Review by yhchiang-sol — DISMISSED** [review]
Good catch.  Thanks for fixing the bug and also improving the test coverage.

One minor comment.

File: ledger/src/blockstore/blockstore_purge.rs (hunk 1 above)
Comment (steviez) [reply ledger/src/blockstore/blockstore_purge.rs]: Looked up the original PR that added this; it seems you intended to use `..=` back then as well 😁 . Changed
https://github.com/solana-labs/solana/pull/27201/files#r960136216

**Review by yhchiang-sol — APPROVED** [review]
LGTM!

**Pull request merged**